### PR TITLE
Turning in week 1 of the gene finder project

### PR DESCRIPTION
**Pull request by NathanLepore**
This was of a reasonable level of difficulty. I feel like I got better at debugging.

### CODE REVIEW

File: gene_finder.py [hunk 1]
```diff
@@ -1,4 +1,4 @@
-# -*- coding: utf-8 -*-
+    # -*- coding: utf-8 -*-
 """
 YOUR HEADER COMMENT HERE
```
Comment (segerphilip) [inline gene_finder.py]: Please include your name in the file and any other needed info (sources and whatnot).

File: gene_finder.py [hunk 2]
```diff
@@ -21,16 +21,17 @@ def shuffle_string(s):
...
+        return "T"
+    if nucleotide == "T":
+        return "A"
+    if nucleotide == "G":
+        return "C"
+    if nucleotide == "C":
+        return "G"
+
+
+# TODO: implement this
```
Comment (segerphilip) [inline gene_finder.py]: After you implement a method, these `#TODO` comments and `pass` can be removed

File: gene_finder.py [hunk 3]
```diff
@@ -82,6 +103,26 @@ def find_all_ORFs_oneframe(dna):
...
+    ORF = list('')
+
+    i = 0
+    while i < length:
+        a = dna[i: i+3]
+        if a == startcodon:
+            dna1 = dna[i:]
+            ORFreturn = rest_of_ORF(dna1)
+            ORFreturn = [ORFreturn]
+            ORF = ORF + ORFreturn
```
Comment (segerphilip) [inline gene_finder.py]: `ORF.append(ORFreturn)` can also be used to add items to a list

File: gene_finder.py [hunk 4]
```diff
@@ -82,6 +103,26 @@ def find_all_ORFs_oneframe(dna):
...
+            i = len(ORFreturn)
```
Comment (segerphilip) [inline gene_finder.py]: Probably should be `i += len(ORFreturn)`

File: gene_finder.py [hunk 5]
```diff
@@ -82,6 +103,26 @@ def find_all_ORFs_oneframe(dna):
...
+        i = i+3
+    return ORF
+
+
+
 
 def find_all_ORFs(dna):
```
Comment (segerphilip) [inline gene_finder.py]: Running tests on this is resulting in Expected: `['ATGCATGAATGTAG', 'ATGAATGTAG', 'ATG']` Actual: `['ATGCATGAATGTAG', 'ATGAATGTAG', 'ATG', 'ATGAATGTAG', 'ATG', 'ATG']`.

Comment (segerphilip) [inline gene_finder.py]: Check how often you are going over the list.

**Comment by NathanLepore** [comment]
Here is my updated gene_finder. It works but it is pretty slow on the salmonella gene.

File: gene_finder.py [hunk 6]
```diff
@@ -1,14 +1,17 @@
...
-@author: YOUR NAME HERE
+@author: Nathan Lepore
 
 """
 
 import random
 from amino_acids import aa, codons, aa_table   # you may find these useful
+
 from load import load_seq
+dna = load_seq("./data/X73525.fa")
```
Comment (segerphilip) [inline gene_finder.py]: You should probably load this in `if __name__ = "__main__":` as that's the perfect area for testing code.

File: gene_finder.py [hunk 7]
```diff
@@ -162,14 +163,24 @@ def find_all_ORFs_both_strands(dna):
     ORF = ORF + find_all_ORFs(dna)
     return ORF
 
+
```
Comment (segerphilip) [inline gene_finder.py]: Is this needed?

File: gene_finder.py [hunk 8]
```diff
@@ -162,14 +163,24 @@ def find_all_ORFs_both_strands(dna):
...
 def longest_ORF(dna):
     """ Finds the longest ORF on both strands of the specified DNA and returns it
         as a string
     >>> longest_ORF("ATGCGAATGTAGCATCAAA")
     'ATGCTACATTCGCAT'
     """
     # TODO: implement this
```
Comment (segerphilip) [inline gene_finder.py]: Once you implement a function you can probably remove these comments

File: gene_finder.py [hunk 9]
```diff
@@ -181,6 +192,15 @@ def longest_ORF_noncoding(dna, num_trials):
         returns: the maximum length longest ORF """
     # TODO: implement this
     pass
```
Comment (segerphilip) [inline gene_finder.py]: and these ones

File: gene_finder.py [hunk 10]
```diff
@@ -209,8 +244,18 @@ def gene_finder(dna):
...
+
+
+print(gene_finder(dna))
+
 
-if __name__ == "__main__":
-    import doctest
-    doctest.run_docstring_examples(find_all_ORFs_both_strands, globals(),
-                                   verbose=True)
+# if __name__ == "__main__":
```
Comment (segerphilip) [inline gene_finder.py]: You should keep this, as it only gets run when you call this specific file. That means it's super useful for testing.

**Pull request closed without merging**